netbox/dcim/api/views.py

```diff
@@ -358,11 +365,15 @@ def graphs(self, request, pk):
 
         return Response(serializer.data)
 
+    @swagger_auto_schema(manual_parameters=[_method], responses={'200': serializers.DeviceNAPALMSerializer})
     @action(detail=True, url_path='napalm')
     def napalm(self, request, pk):
         """
         Execute a NAPALM method on a Device
         """
+        if not request.GET.get('method'):
```

Comment:
I see why you did this, but this is technically out of scope for #1982 as it modifies the endpoint behavior. It needs a separate issue; I'd argue for a bug report (rather than a FR).

I say this because in a scenario where a user has a script making malformed calls to the NAPALM endpoint, this change might cause it to start throwing errors. We want to make sure the user can quickly discern the suspect reason from reviewing the release notes.

Reply:
Took it out. To be honest, I'm not at all bothered by the fact that it doesn't do anything. I just misunderstood the original ticket when the person was referring to the call not "running anything".